src/overlays/actors/ovl_En_Takaraya/z_en_takaraya.c

```diff
@@ -28,61 +47,360 @@ ActorInit En_Takaraya_InitVars = {
+                    Animation_MorphToLoop(&this->skelAnime, &object_bg_Anim_009890, 5.0f);
+                }
+            }
+        } else {
+            Audio_PlaySfx_MessageCancel();
+            this->actor.textId = D_80ADFB44[(void)0, gSaveContext.save.playerForm];
+            if (this->skelAnime.animation == &object_bg_Anim_009890) {
+                Animation_MorphToPlayOnce(&this->skelAnime, &object_bg_Anim_000968, 5.0f);
+            }
+        }
+        Message_ContinueTextbox(play, this->actor.textId);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Takaraya/EnTakaraya_Init.s")
+void func_80ADF2D4(EnTakaraya* this) {
+    Actor_PlaySfx_FlaggedCentered2(&this->actor, NA_SE_SY_FOUND);
+    Audio_PlaySubBgm(NA_BGM_TIMED_MINI_GAME);
+    this->timer = 145;
+    SET_WEEKEVENTREG(WEEKEVENTREG_KICKOUT_WAIT);
+    CLEAR_WEEKEVENTREG(WEEKEVENTREG_KICKOUT_TIME_PASSED);
+    this->actionFunc = &func_80ADF338;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Takaraya/EnTakaraya_Destroy.s")
+void func_80ADF338(EnTakaraya* this, PlayState* play) {
+    Actor* chest = play->actorCtx.actorLists[ACTORCAT_CHEST].first;
+    Actor* player = play->actorCtx.actorLists[ACTORCAT_PLAYER].first;
+    Vec3f eye;
+    Vec3f at;
+    f32 sp2C;
+    s16 CamId;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Takaraya/func_80ADED8C.s")
+    SkelAnime_Update(&this->skelAnime);
+    if ((CutsceneManager_GetCurrentCsId() == this->actor.csId) && (chest != NULL)) {
+        this->timer--;
+        CamId = CutsceneManager_GetCurrentSubCamId(this->actor.csId);
+        if (this->timer >= 26) {
+            sp2C = 250.0f;
+        } else {
+            sp2C = ((chest->xzDistToPlayer - 250.0f) * (25 - this->timer) * 0.04f) + 250.0f;
+        }
+        eye.x = (Math_SinS(chest->yawTowardsPlayer) * sp2C) + chest->world.pos.x;
+        eye.y = player->world.pos.y + 120.0f;
+        eye.z = (Math_CosS(chest->yawTowardsPlayer) * sp2C) + chest->world.pos.z;
+        at.x = eye.x - (Math_SinS(chest->yawTowardsPlayer) * 250.0f);
+        at.y = eye.y - 90.0f;
+        at.z = eye.z - (Math_CosS(chest->yawTowardsPlayer) * 250.0f);
+        Play_SetCameraAtEye(play, CamId, &at, &eye);
+    } else if (this->timer < 145) {
+        func_80ADF4E0(this);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Takaraya/func_80ADEDF8.s")
+void func_80ADF4E0(EnTakaraya* this) {
+    Interface_StartTimer(TIMER_ID_MINIGAME_2, 45);
+    this->timer = 1;
+    this->actionFunc = &func_80ADF520;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Takaraya/func_80ADEE4C.s")
+void func_80ADF520(EnTakaraya* this, PlayState* play) {
+    SkelAnime_Update(&this->skelAnime);
+    if (Play_InCsMode(play) == 0) {
+        if (Flags_GetTreasure(play, this->actor.params)) {
+            Flags_SetSwitch(play, this->unk2B0);
+            play->actorCtx.sceneFlags.chest &= ~(1 << this->actor.params);
```

Comment:
this->actor.params could use a `TAKARAYA_GET_TREASURE_FLAG` macro by the looks of it.